windows/src/engine/keyman32/appcontext.cpp

```diff
@@ -164,3 +164,36 @@ ContextItemToAppContext(km_core_context_item *contextItems, PWSTR outBuf, DWORD
   delete[] buf;
   return TRUE;
 }
+
+FormatContextResult
+format_context_for_core(LPCWSTR windows_context, LPWSTR core_context, uint32_t output_size) {
+  if (windows_context == nullptr || core_context == nullptr) {
+    return frERROR;
+  }
+  // Return early if windowsContext does not contain '\r\n'
+  if (wcsstr(windows_context, L"\r\n") == nullptr) {
+    return frNO_CHANGE;
+  }
+
+  auto windows_context_length = wcsnlen_s(windows_context, MAXCONTEXT);
+
+  if (output_size < windows_context_length) {
```

Comment:
```suggestion
  if (output_size <= windows_context_length) {
```

Need space for terminating nul